src/mimeheaders.rs

```diff
@@ -24,36 +25,36 @@ pub struct MimeContentTypeHeader {
 }
 
 impl FromHeader for MimeContentTypeHeader {
-    fn from_header(value: String) -> Option<MimeContentTypeHeader> {
+    fn from_header(value: String) -> ParsingResult<MimeContentTypeHeader> {
         let mut parser = Rfc2045Parser::new(value.as_slice());
         let (value, params) = parser.consume_all();
 
         let mime_parts: Vec<&str> = value.as_slice().splitn(2, '/').collect();
 
         if mime_parts.len() == 2 {
-            Some(MimeContentTypeHeader {
+            Ok(MimeContentTypeHeader {
                 content_type: (mime_parts[0].to_string(), mime_parts[1].to_string()),
                 params: params
             })
         } else {
-            None
+            Err(ParsingError::new("Too many mime parts.".to_string()))  // FIXME
```

Comment:
Could we just panic here, or don't do the check? If ``splitn(2, ...)`` returns more than two items, there must be something else wrong.